shared/utils/accessor-wrapper-types.hpp

```diff
@@ -37,54 +63,111 @@ namespace bs_hook {
     struct InstanceProperty;
     
     template<internal::NTTPString name, class T>
     struct InstanceProperty<name, T, true, false> {
-        explicit InstanceProperty(void* inst) noexcept : instance(inst) {}
+        explicit constexpr InstanceProperty(void* inst) noexcept : instance(inst) {}
         operator T() const {
             auto res = il2cpp_utils::GetPropertyValue<T, false>(reinterpret_cast<Il2CppObject*>(const_cast<void*>(instance)), name.data.data());
             if (!res) throw PropertyException(std::string("Failed to get instance property: ") + name.data.data());
             return *res;
         }
+
+        inline auto operator ->() const {
+            return this->operator T();
+        } 
+
+        inline auto operator *() const {
+            return this->operator T();
+        }
+
+        inline auto v() const {
+            return this->operator T();
+        }
+
+        template<typename... Targs>
+        decltype(auto) operator ()(Targs&&... args) { return this->operator T()(std::forward<Targs...>(args...)); }
+
         private:
         void* instance;
     };
 
     template<internal::NTTPString name, class T>
     struct InstanceProperty<name, T, false, true> {
-        explicit InstanceProperty(void* inst) noexcept : instance(inst) {}
+        explicit constexpr InstanceProperty(void* inst) noexcept : instance(inst) {}
         InstanceProperty& operator=(T&& t) {
             auto val = reinterpret_cast<Il2CppObject*>(instance);
             auto res = il2cpp_utils::SetPropertyValue<false>(val, name.data.data(), std::forward<decltype(t)>(t));
             if (!res) throw PropertyException(std::string("Failed to set instance property: ") + name.data.data());
             return *this;
         }
+        
         private:
         void* instance;
     };
 
+#define BINARY_OPERATOR_OP_EQ_PROP(op)      \
+template<typename U>                        \
+auto& operator op##=(const U& rhs) {        \
+    auto temp = this->operator T();         \
+    return this->operator=(temp op##= rhs); \
+}
+
     template<internal::NTTPString name, class T>
     struct InstanceProperty<name, T, true, true> {
-        explicit InstanceProperty(void* inst) noexcept : instance(inst) {}
+        explicit constexpr InstanceProperty(void* inst) noexcept : instance(inst) {}
         operator T() const {
             auto res = il2cpp_utils::GetPropertyValue<T, false>(reinterpret_cast<Il2CppObject*>(const_cast<void*>(instance)), name.data.data());
             if (!res) throw PropertyException(std::string("Failed to get instance property: ") + name.data.data());
             return *res;
         }
-        InstanceProperty& operator=(T&& t) {
+        InstanceProperty& operator=(T const& t) {
             auto val = reinterpret_cast<Il2CppObject*>(instance);
             auto res = il2cpp_utils::SetPropertyValue<false>(val, name.data.data(), std::forward<decltype(t)>(t));
```

Comment:
This should no longer forward, I think (ditto others in this file that do the same) and instead strip the `const` for the set, since it expects an rvalue reference originally. Allowing copies of T here is interesting, I'm not sure how I feel about it instead of forcing moves though...

Reply:
honestly idk what you're even saying

Reply:
Basically, because you swapped T&& with T const&, you don't need the std::forward. I think it's fine to do it this way